lib/excon.rb

```diff
@@ -128,7 +128,12 @@ def new(url, params = {})
       if uri.user
         params[:user] = Utils.unescape_uri(uri.user)
       end
-      Excon::Connection.new(params)
+      begin
+          con = Excon::Connection.new(params)
+      rescue Exception => ex
+          print "Unable to establish Excon connection.\nException detail:\n#{ex}\nDid you set your proxy environment variable correctly?\n\n"
```

Comment:
If we fix the stuff below, this seems like it might be redundant? What do you think?